src/components/dropdown/daikin-dropdown.ts

```diff
@@ -0,0 +1,319 @@
+import type { MergeVariantProps } from "../../type-utils";
+import type { DaikinDropdownItem } from "../dropdown-item";
+import "../icon/daikin-icon";
+import type { IconType } from "../icon/daikin-icon";
+
+type SelectEvent = Event & {
+  detail: { value: string };
+};
+
+const cvaContainer = cva(["flex", "gap-2", "w-max", "relative"], {
+  variants: {
+    labelPosition: {
+      top: ["flex-col", "gap-2"],
+      left: ["items-center", "gap-3"],
+      hidden: [],
+    },
+  },
+});
+
+const cvaLabel = cva([], {
+  variants: {
+    labelPosition: {
+      top: [],
+      left: [],
+      hidden: ["hidden"],
+    },
+  },
+});
+
+const cvaButton = cva(
+  [
+    "flex",
+    "items-center",
+    "gap-2",
+    "w-[218px]",
+    "bg-white",
+    "border",
+    "border-[#8c8c8c]",
+    "rounded-md",
+    "overflow-hidden",
+    "font-daikinSerif",
+    "text-left",
+
+    "hover:bg-[#ebebeb]",
+
+    "relative",
+    "after:i-daikin-dropdown-chevron-down",
+    "after:w-3",
+    "after:h-3",
+    "after:absolute",
+    "after:m-auto",
+    "after:top-0",
+    "after:bottom-0",
+    "after:right-3",
+  ],
+  {
+    variants: {
+      size: {
+        small: ["min-h-8", "py-1", "px-2", "text-[13px]", "leading-5"],
+        medium: [
+          "min-h-[42px]",
+          "py-2.5",
+          "px-3",
+          "text-[15px]",
+          "leading-[22px]",
+        ],
+      },
+    },
+  }
+);
+
+const cvaContent = cva(
+  [
+    "w-max",
+    "border",
+    "border-[#8c8c8c]",
+    "rounded-md",
+    "overflow-hidden",
+    "absolute",
+    "top-0",
+    "left-0",
+    "opacity-1",
+    "transition-[opacity]",
+  ],
+  {
+    variants: {
+      state: {
+        visible: [],
+        hidden: ["opacity-0", "pointer-events-none"],
+      },
+    },
+  }
+);
+
+type DropdownVariantProps = MergeVariantProps<
+  typeof cvaContainer | typeof cvaButton
+>;
+
+/**
+ * * A dropdown list component.
+ *
+ * @example
+ *
+ * ```html
+ * <daikin-dropdown>
+ *   <daikin-dropdown-item value="value1">Item 1</daikin-dropdown-item>
+ *   <daikin-dropdown-item value="value2">Item 2</daikin-dropdown-item>
+ *   <daikin-dropdown-item value="value3">Item 3</daikin-dropdown-item>
+ * </daikin-dropdown>
+ * ```
+ */
+@customElement("daikin-dropdown")
+export class DaikinDropdown extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: block;
+      width: max-content;
+    }
+
+    ::slotted(daikin-dropdown-item:not(:last-child)) {
+      border-bottom: 1px solid #8c8c8c;
+    }
+  `;
+
+  /**
+   * Label text
+   */
+  @property({ type: String })
+  label: string = "";
+
+  /**
+   * Specify the size of the dropdown
+   */
+  @property({ type: String })
+  size: DropdownVariantProps["size"] = "medium";
+
+  /**
+   * Where the label is located in terms of the dropdown
+   */
+  @property({ type: String, reflect: true, attribute: "label-position" })
```

Comment:
Basically, I want to decide whether or not to reflect based on the following policy:

- If it is Boolean, then reflect
- If it is a String, reflect if there is a possibility of using an attribute selector.

```suggestion
  @property({ type: String, attribute: "label-position" })
```